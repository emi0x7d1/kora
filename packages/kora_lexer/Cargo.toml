[package]
name = "kora_lexer"
version = "0.1.0"
edition = "2021"

[dependencies]
unicode-ident = "1.0.11"

[dev-dependencies]
insta = { version = "1.31.0", features = ["glob"]}
