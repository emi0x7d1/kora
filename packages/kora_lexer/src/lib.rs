#![allow(unused)]

mod error;
mod lexer;
mod macros;
mod token;

pub use lexer::Lexer;
pub use token::{Token, TokenKind};
pub use error::SyntaxError;
