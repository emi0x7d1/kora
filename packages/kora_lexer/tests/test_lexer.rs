use kora_lexer::Lexer;

#[test]
fn test_lexer() {
    insta::glob!("inputs/*.kora", |path| {
        let input = std::fs::read_to_string(path).unwrap();

        let tokens = Lexer::new(&input).collect::<Vec<_>>();

        insta::with_settings!({
            description => &input,
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(tokens);
        });
    })
}
