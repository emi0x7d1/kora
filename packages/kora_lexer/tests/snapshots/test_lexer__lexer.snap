---
source: packages/kora_lexer/tests/test_lexer.rs
description: "def extend with if else for struct an_ídentifier\n"
input_file: packages/kora_lexer/tests/inputs/keywords.kora
---
[
    Token {
        kind: Def,
        text: "def",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: Extend,
        text: "extend",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: With,
        text: "with",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: If,
        text: "if",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: Else,
        text: "else",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: For,
        text: "for",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: Struct,
        text: "struct",
    },
    Token {
        kind: Trivia,
        text: " ",
    },
    Token {
        kind: Identifier,
        text: "an_ídentifier",
    },
    Token {
        kind: Trivia,
        text: "\n",
    },
]
